### PR TITLE
add fast coco eval api

### LINKED ISSUE #163: add optional fast coco api
**If this is your first time, please read our contributor guidelines:** https://gitee.com/mindspore/mindspore/blob/master/CONTRIBUTING.md

**Is your feature request related to a problem? Please describe.**
original coco map calculation api is too slow

**Describe the solution you'd like**
add optional fast coco api
**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Mark-ZhouWX**
Thank you for your contribution to the MindYOLO repo.
Before submitting this PR, please make sure:

- [x] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindyolo/blob/master/CONTRIBUTING.md)
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved terminology
- [ ] You have added unit tests

## Motivation

(Write your motivation for proposed changes here.)
1. to speed up coco map calculation

## Test Plan

(How should this PR be tested? Do you require special setup to run the test or repro the fixed bug?)

## Related Issues and PRs
close #163 
(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

File: mindyolo/csrc/build.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
```
Comment (CaitinZhao) [inline mindyolo/csrc/build.sh:1]: 这个在install的文档里也说下，顺便补充下whl包安装和源码安装两种方式

File: test.py [hunk 2]
```diff
@@ -11,7 +11,11 @@
 import numpy as np
 from mindspore.communication import init, get_rank, get_group_size
 from pycocotools.coco import COCO
-from pycocotools.cocoeval import COCOeval
+try:
+    from mindyolo.csrc import COCOeval_fast as COCOeval
+except ImportError as e:
+    print(f'[Warning] unable to load fast_coco_eval api, use normal one instead')
```
Comment (CaitinZhao) [inline test.py:17]: 直接用log打吧

Comment (Mark-ZhouWX) [reply test.py:17]: logger在set_default_test里初始化，import这句的时候还没初始化

**Review by zhanghuiyao — DISMISSED** [review]

File: mindyolo/csrc/build.sh (hunk 1 above)
Comment (Mark-ZhouWX) [reply mindyolo/csrc/build.sh:1]: 已完成

**Review by zhanghuiyao — DISMISSED** [review]

File: README.md [hunk 3]
```diff
@@ -53,12 +53,29 @@ See [MODEL ZOO](MODEL_ZOO.md).
 - pyyaml >= 5.3
 - openmpi 4.0.3 (for distributed mode)
 
-To install the dependency, please run
+### Case A:
+Install mindyolo as a package, please run:
```
Comment (CaitinZhao) [inline README.md:57]: 最优先

pip install mindyolo --trusted-host ms-release.obs.cn-north-4.myhuaweicloud.com -i https://pypi.tuna.tsinghua.edu.cn/simple

也可以 clone 源码安装whl包

File: README.md [hunk 4]
```diff
@@ -53,12 +53,29 @@ See [MODEL ZOO](MODEL_ZOO.md).
 - pyyaml >= 5.3
 - openmpi 4.0.3 (for distributed mode)
 
-To install the dependency, please run
+### Case A:
```
Comment (CaitinZhao) [inline README.md:56]: ## Installation

### Case A

Install mindyolo as a package, please run:

```shell
pip install mindyolo --trusted-host ms-release.obs.cn-north-4.myhuaweicloud.com -i https://pypi.tuna.tsinghua.edu.cn/simple
```

or

```shell
pip install git+https://github.com/mindspore-lab/mindyolo.git
```

### Case B

Install mindyolo by Source Code：

```shell
git clone https://github.com/mindspore-lab/mindyolo.git
cd mindyolo
pip install -r requirements.txt
```

It is suggested to build fast coco api extension for better performance in evaluating coco data. Note that this behavior is optional:

```shell
cd $project_root/mindyolo/csrc
sh build.sh
```

**Review by CaitinZhao — APPROVED** [review]

**Review by zhanghuiyao — APPROVED** [review]

**Pull request merged**